Core/include/Acts/Surfaces/Surface.hpp

```diff
@@ -310,14 +290,11 @@ class Surface : public virtual GeometryObject,
   /// @param position global 3D position - considered to be on surface but not
   /// inside bounds (check is done)
   /// @param momentum global 3D momentum representation (optionally ignored)
-  /// @param lposition local 2D position to be filled (given by reference for
-  /// method symmetry)
   ///
-  /// @return boolean indication if operation was successful (fail means global
-  /// position was not on surface)
-  virtual bool globalToLocal(const GeometryContext& gctx,
-                             const Vector3D& position, const Vector3D& momentum,
-                             Vector2D& lposition) const = 0;
+  /// @return a Result<Vector2D> which can be !ok() if the operation fails
+  virtual Result<Vector2D> globalToLocal(const GeometryContext& gctx,
```

Comment:
I like this! Only consideration is: if it fails, you will not be able to access the result value (even if it's just *incorrect* mathematically). Is that acceptable?

Reply:
I think so, it gives you a clean understanding what happened. 
If you want to just bring any position into a Surface reference frame, you better do it by hand through the `Surface::transform(...)`

Reply:
But that does not work for non-planar surfaces, right? The transform only computes the local Euclidean part but does not handle e.g. the conversion to polar coordinates.

Reply:
This might be a question of policy: we can either return `Result<...>` and thus ensure that the local position **must** be on the surface, or we can return a regular `Vector2D` and then require that the returned position is e.g. the closest point on the surface if the input point is not **on** the surface, or we can provide a separate method that returns the full 3d vector in the local frame where the third component is zero, iff the position is on the surface.

Reply:
I would personally say that having it fail explicitly by default is the way to go. In that case, however, we should provide a way to circumvent this if need be, such as another function which operates in the way @msmk0 describes, returning a 3D verctor.